MyString.java

```diff
@@ -78,23 +116,43 @@ public static String randomStringOfLetters(int n) {
      * @return a string consisting of str1 minus all the characters of str2
      */
     public static String remove(String str1, String str2) {
-       //// Replace the following statement with your code
-        return null;
-    }
-
-    /**
-     * Returns a string consisting of the given string, with the given 
-     * character inserted randomly somewhere in the string.
-     * For example, insertRandomly("s","cat") can return "scat", or "csat", or "cast", or "cats".  
-     * @param ch - a character
-     * @param str - a string
-     * @return a string consisting of str with ch inserted somewhere
-     */
-    public static String insertRandomly(char ch, String str) {
-         // Generate a random index between 0 and str.length()
-         int randomIndex = (int) (Math.random() * (str.length() + 1));
-         // Insert the character at the random index
-         String result = str.substring(0, randomIndex) + ch + str.substring(randomIndex);
-         return result;
-    }    
-}
+        char[] newCharArray= new char[str1.length()];
+        for (int i=0; i<str1.length(); i++){
+            newCharArray[i]=str1.charAt(i);
+        }
+        for (int j=0; j<str2.length(); j++){
+            char char2= str2.charAt(j);
+            for (int k=0; k<newCharArray.length; k++){
+                if (newCharArray[k]==char2 || newCharArray[k]==char2-32 || newCharArray[k]==char2+32){
+                    newCharArray[k]=' ';
+                    break;

```

Comment:
32 is magic number
use formula..
this is also not good logically.. better to use lower/upper case and stick to 1